iina/PlayerWindowController.swift

```diff
@@ -255,11 +258,51 @@ class PlayerWindowController: NSWindowController, NSWindowDelegate {
 
   override func keyDown(with event: NSEvent) {
     let keyCode = KeyCodeHelper.mpvKeyCode(from: event)
-    if let kb = PlayerCore.keyBindings[keyCode] {
+    if keyCode != "" {
+      #if DEBUG
+      Logger.log("KeyDown: \(keyCode)", level: .verbose)
+      #endif
+      if matchAndHandle(keyCode) {
+        return
+      }
+
+      // try to match key sequences, up to 4 values. shortest match wins
+      var keyCodeSequence = keyCode
+      for i in 0..<3 {
+        let prevKeyCode = lastKeysPressed[(lastKeyPressedIndex+4-i)%4]
```

Comment:
Remember I'm the newbie. Might want to wait and see what @lhc70000 thinks...

Instead of implementing our own ring buffer, maybe it would be easier to understand if the code used a Deque from swift-collections. Something like:
```swift
internal var lastKeysPressed = Deque<String>()

private func rememberKeyPress(_ keyCode: String) {
  if lastKeysPressed.count == 4 {
    lastKeysPressed.removeLast()
  }
  lastKeysPressed.prepend(keyCode)
}
```
And:
```swift
 rememberKeyPress("a")
 rememberKeyPress("b")
 var keyCodeSequence = "c"
 var iterator = lastKeysPressed.makeIterator()
 while let prevKeyCode = iterator.next() {
   keyCodeSequence = "\(prevKeyCode)-\(keyCodeSequence)"
   print(keyCodeSequence)
 }
```
Something along those lines...

Reply:
I agree with the concern, although perhaps not the implementation. What we really want here is a fixed-size buffer, which your array is, but inlining the details here isn't great. @svobs, would you be up for writing a RingBuffer Collection and sticking it in our utilities folder, so we can reuse it for other things too?

Reply:
A RingBuffer Collection in utilities like @saagarjha suggests sounds good to me.